model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/base_gumbel_rounding.py

```diff
@@ -31,21 +31,29 @@
 P_INIT = 0.01
 
 
-def _init_aux_var(w_shape: List[int], m: int, p: float = P_INIT) -> np.ndarray:
+def init_aux_var(ceil_indicator: np.ndarray, w_shape: List[int], m: int, p: float = P_INIT) -> np.ndarray:
     """
     This function generate a random pi matrix for Gumbel Rounding
     Args:
-        w_shape(List[int]): A list of integers that represent the shape of the weights tensor to be quantization
+        ceil_indicator: an array of indicator if the value should be ceil or floor.
+        w_shape(List[int]): A list of integers that represent the shape of the weights tensor to be quantization.
         p(float): A floating point number that represent the probability of non round options of pi matrix.
-        m(int):  An integer that define the number of shift
+        m(int):  An integer that define the number of shift.
 
     Returns: A numpy array of pi tensor
 
     """
-    m_hat = m // 2
+    if m < 2:
+        common.logger.Logger.error("m must be larger than two")
+    if m % 2 != 0:
+        common.logger.Logger.error("m must be module two")
+    m_hat = m // 2 - 1
```

Comment:
If m must be dividable by 2 then I think there is no need for floor division (it's not critical)